client/src/components/App.jsx

```diff
@@ -67,6 +69,32 @@ class App extends React.Component {
     this.setState({ currentStyleID: id });
   }
 
+  addProductToFavorites() {
+    let productToAdd = this.state.currentProductID;
+    if (!this.favorites[productToAdd]) {
+      this.favorites[productToAdd] = true;
+      let productInfo = this.cachedProducts[productToAdd];
+      let updatedFavoriteProducts = this.state.favoriteProducts.slice();
+      updatedFavoriteProducts.push(productInfo);
+      this.setState({
+        favoriteProducts: updatedFavoriteProducts
+      });
+    }
+  }
```

Comment:
logic is easy to understand. Nice functionality

Reply:
Thanks Alexander =)